### PR TITLE
Replace DynamoImage class with DynamoEvent

### PR DESCRIPTION
**Pull request by kenoir**
In order to more closely mirror dynamo API - this will enable us to notice delete events more effectively.

### CODE REVIEW

**Comment by alexwlchan** [comment]
Assuming this isn’t super critical, can we go through this tomorrow?

**Comment by kenoir** [comment]
@alexwlchan ~~i'd like to make this available as soon as possible as i've got a use case where I need to see deleted records (which this should provide). If you see a critical issue here let me know, otherwise i'd like to merge it.~~ Strike that, I can wait :)

File: src/wellcome_aws_utils/dynamo_event.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+from enum import Enum
+
+from boto3.dynamodb.types import TypeDeserializer
+
+
+class DynamoEventFactory:
+    @staticmethod
+    def create(event):
+        return [DynamoEvent(record) for record in event['Records']]
```
Comment (alexwlchan) [inline src/wellcome_aws_utils/dynamo_event.py]: Make this a function and a generator.

File: src/wellcome_aws_utils/dynamo_event.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    REMOVE, INSERT, MODIFY = range(3)
+
+
+class DynamoEvent:
+    def _sanity_check_record(self, record):
+        if not ('eventSource' in record):
+            raise Exception(f'Unrecognised event: {record}')
+
+        if record['eventSource'] != 'aws:dynamodb':
+            raise Exception(f'Event source is not aws:dynamodb: {record}')
```
Comment (alexwlchan) [inline src/wellcome_aws_utils/dynamo_event.py]: EAFP instead of LBYL.

Comment (alexwlchan) [reply src/wellcome_aws_utils/dynamo_event.py]: And use UnWellcomeException.

File: src/wellcome_aws_utils/dynamo_event.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    def _set_event_type(self, record):
+        if 'eventName' not in record:
+            raise Exception(f'No eventName found in {record}!')
+
+        self.event_type = None
+        if self.record['eventName'] == 'REMOVE':
+            self.event_type = DynamoEventType.REMOVE
+
+        if self.record['eventName'] == 'INSERT':
```
Comment (alexwlchan) [inline src/wellcome_aws_utils/dynamo_event.py]:
```python
if self.record['eventName'] == 'REMOVE':
    ...
elif self.record['eventName'] == 'INSERT':
    ...
elif ...:
    ...
else:
    ...
```

File: src/wellcome_aws_utils/dynamo_event.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+            raise Exception(
+                f'No Keys attribute available on record.dynamodb: {dynamodb}'
+            )
+
+        self._keys = dynamodb['Keys']
+
+    def _set_images(self, dynamodb):
+        new_image = None
+        if 'NewImage' in dynamodb:
+            new_image = dynamodb['NewImage']
```
Comment (alexwlchan) [inline src/wellcome_aws_utils/dynamo_event.py]:
```python
new_image = dynamodb.get(key, default=None)
```

```python
new_image = dynamodb.get('NewImage')
```

**Comment by kenoir** [comment]
@alexwlchan hopefully addressed review comments.

**Pull request merged**